lib/Base/Tensor.cpp

```diff
@@ -599,28 +599,14 @@ Tensor Tensor::getCopyConvertedToType(ElemKind newKind) const {
              {dims()[0], dims()[1] - 2 * (sizeof(float) - sizeof(float16_t))},
              1.0, 0);
 
-  const size_t srcWidth = dims()[1];
   const size_t dstWidth = tmp.dims()[1];
-  auto *srcData = reinterpret_cast<uint8_t *>(getUnsafePtr());
-  auto *dstData = reinterpret_cast<uint8_t *>(tmp.getUnsafePtr());
   auto srcH = getHandle<uint8_t>();
   auto dstH = tmp.getHandle<uint8_t>();
   for (size_t i = 0, e = dims()[0]; i < e; i++) {
-    // Get the src's scale/offset for the row.
-    uint8_t *srcScaleOffsetPtr =
-        &srcData[(i + 1) * srcWidth] - 2 * sizeof(float);
-    float srcScale, srcOffset;
-    memcpy(&srcScale, srcScaleOffsetPtr, sizeof(float));
-    memcpy(&srcOffset, srcScaleOffsetPtr + sizeof(float), sizeof(float));
-
-    // Convert scale/offset to FP16 and copy only those to dst.
-    float16_t dstScale = static_cast<float16_t>(srcScale);
-    float16_t dstOffset = static_cast<float16_t>(srcOffset);
-    uint8_t *dstScaleOffsetPtr =
-        &dstData[(i + 1) * dstWidth] - 2 * sizeof(float16_t);
-    memcpy(dstScaleOffsetPtr, &dstScale, sizeof(float16_t));
-    memcpy(dstScaleOffsetPtr + sizeof(float16_t), &dstOffset,
-           sizeof(float16_t));
+    // Copy the scale/offset from src to dst.
+    float scale, offset;
+    std::tie(scale, offset) = srcH.getFusedScaleOffsetFromRow<float>(i);
+    dstH.setFusedScaleOffsetInRow<float16_t>(i, scale, offset);
```

Comment:
nit: is it better to use static_cast here?  `scale->static_cast<float16_t>scale`?

Reply:
Hm I think the implicit conversion will do the cast as expected, but I can add it in to be more explicit.